com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -432,163 +494,141 @@ void Send(NetworkTransformState stateToSend)
+                ReplicatedNetworkState.Value = m_LocalAuthoritativeNetworkState;
                 m_HasSentLastValue = false;
                 m_LastSentTick = m_CachedNetworkManager.LocalTime.Tick;
                 m_LastSentState = m_LocalAuthoritativeNetworkState;
             }
             else if (!m_HasSentLastValue && m_CachedNetworkManager.LocalTime.Tick >= m_LastSentTick + 1) // check for state.IsDirty since update can happen more than once per tick. No need for client, RPCs will just queue up
             {
+                m_LastSentState.IsTeleportingNextFrame = false; // This is required here
                 m_LastSentState.SentTime = m_CachedNetworkManager.LocalTime.Time; // time 1+ tick later
-                Send(m_LastSentState);
+                // Commit the state
+                ReplicatedNetworkState.Value = m_LastSentState;
                 m_HasSentLastValue = true;
             }
         }
 
-        [ServerRpc(RequireOwnership = false)]
-        private void CommitTransformServerRpc(NetworkTransformState networkState, ServerRpcParams serverParams = default)
-        {
-            if (serverParams.Receive.SenderClientId == OwnerClientId) // RPC call when not authorized to write could happen during the RTT interval during which a server's ownership change hasn't reached the client yet
-            {
-                CommitLocallyAndReplicate(networkState);
-            }
-        }
-
-        private void CommitLocallyAndReplicate(NetworkTransformState networkState)
-        {
-            m_ReplicatedNetworkState.Value = networkState;
-
-            if (Interpolate)
-            {
-                AddInterpolatedState(networkState);
-            }
-        }
-
         private void ResetInterpolatedStateToCurrentAuthoritativeState()
         {
             var serverTime = NetworkManager.ServerTime.Time;
             m_PositionXInterpolator.ResetTo(m_LocalAuthoritativeNetworkState.PositionX, serverTime);
             m_PositionYInterpolator.ResetTo(m_LocalAuthoritativeNetworkState.PositionY, serverTime);
             m_PositionZInterpolator.ResetTo(m_LocalAuthoritativeNetworkState.PositionZ, serverTime);
 
-            m_RotationInterpolator.ResetTo(Quaternion.Euler(m_LocalAuthoritativeNetworkState.Rotation), serverTime);
+            m_RotationInterpolator.ResetTo(Quaternion.Euler(m_LocalAuthoritativeNetworkState.RotAngleX, m_LocalAuthoritativeNetworkState.RotAngleY, m_LocalAuthoritativeNetworkState.RotAngleZ), serverTime);
 
             m_ScaleXInterpolator.ResetTo(m_LocalAuthoritativeNetworkState.ScaleX, serverTime);
             m_ScaleYInterpolator.ResetTo(m_LocalAuthoritativeNetworkState.ScaleY, serverTime);
             m_ScaleZInterpolator.ResetTo(m_LocalAuthoritativeNetworkState.ScaleZ, serverTime);
         }
 
         /// <summary>
-        /// Will apply the transform to the LocalAuthoritativeNetworkState and get detailed isDirty information returned.
+        /// Used for integration testing:
```

Comment:
Any time I see a comment like this, I wonder if this method should be here. Could the tests replace NetworkTransform with a subclass that adds this method?

Reply:
That was there prior to the update.  I just adjusted the comment to call it out for that purpose.
I added that to the list of things we can continue to improve upon for NetworkTransform.

Reply:
👍 Recording a todo for later is good enough for now.